File: q0057/q0057.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        mergedIntervals = []
        for currInterval in intervals: 
            if newInterval[0] > currInterval[1]:
                mergedIntervals.append(currInterval)
            elif newInterval[1] < currInterval[0]:
                mergedIntervals.append(newInterval)
                newInterval = currInterval
            else: 
                newInterval[0] = min(newInterval[0], currInterval[0])
                newInterval[1] = max(newInterval[1], currInterval[1])
        mergedIntervals.append(newInterval)
        return mergedIntervals

File: q0057/q0057.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        List<int[]> mergedIntervals = new ArrayList<>();

        for (int[] currInterval : intervals) {
            // If the current interval ends before the new interval
            if (currInterval[1] < newInterval[0]) {
                // Add the current interval and move on
                mergedIntervals.add(currInterval);
            }

            // If the current interval ends after the new interval
            else if (currInterval[0] > newInterval[1]) {
                // Add the new interval and set the current interval as new interval
                mergedIntervals.add(newInterval);
                newInterval = currInterval;
            }

            // If the current interval has interference with the new interval
            else {
                // Merge the intervals
                newInterval[0] = Math.min(currInterval[0], newInterval[0]);
                newInterval[1] = Math.max(currInterval[1], newInterval[1]);
            }
        }
        mergedIntervals.add(newInterval);

        return mergedIntervals.toArray(new int[mergedIntervals.size()][]);
    }
}